services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/tenant/ManagementStore.java

```diff
@@ -443,4 +452,40 @@ protected Future<UpdateResult> updateJsonField(
                 checkSpan -> readTenantEntryById(this.client, tenantId, checkSpan.context()));
     }
 
+    /**
+     * Gets a list of tenants.
+     * @param pageSize the page size
+     * @param pageOffset the page offset
+     * @param spanContext The span to contribute to.
+     *
+     * @return A future containing tenants
+     */
+    public Future<SearchResult<TenantWithId>> find(final int pageSize, final int pageOffset, final SpanContext spanContext) {
+
+        final var expanded = this.findTenantsStatement.expand(map -> {
+            map.put("page_size", pageSize);
+            map.put("page_offset", pageOffset);
+        });
+
+        final Span span = TracingHelper.buildChildSpan(this.tracer, spanContext, "find tenants", getClass().getSimpleName())
+            .start();
+
+        return expanded
+            .trace(this.tracer, span.context())
+            .query(this.client)
+            .map(r -> {
+                final var entries = r.getRows(true);
+                span.log(Map.of(
+                    "event", "read result",
+                    "rows", entries.size()));
+                final List<TenantWithId> list = new ArrayList<>();
+                for (var entry : entries) {
+                    final var id = entry.getString("tenant_id");
+                    final var tenant = Json.decodeValue(entry.getString("data"), Tenant.class);
+                    list.add(TenantWithId.from(id, tenant));
+                }
+                return new SearchResult<>(list.size(), list);
```

Comment:
the `list.size()` will only contain the number of tenants on the page, right? If so, then we also need a COUNT query here ...

Reply:
correct... done